.travis.yml

```diff
@@ -1,44 +1,59 @@
-          exit
-        }
-      fi
+- |-
+    case $TRAVIS_OS_NAME in
+      windows)
+        [[ ! -f C:/tools/msys64/msys2_shell.cmd ]] && rm -rf C:/tools/msys64
+        choco uninstall -y mingw
+        choco upgrade --no-progress -y msys2
+        export msys2='cmd //C RefreshEnv.cmd '
+        export msys2+='& set MSYS=winsymlinks:nativestrict '
+        export msys2+='& C:\\tools\\msys64\\msys2_shell.cmd -defterm -no-start'
+        export mingw64="$msys2 -mingw64 -full-path -here -c \$\* --"
+        export msys2+=" -msys2 -c \$\* --"
+        $msys2 pacman --sync --noconfirm --needed mingw-w64-x86_64-toolchain autoconf libtool automake make autoconf-archive pkg-config gcc
+        ## Install more MSYS2 packages from https://packages.msys2.org/base here
+        taskkill //IM gpg-agent.exe //F  # https://travis-ci.community/t/4967
+        export PATH=/C/tools/msys64/mingw64/bin:$PATH
+        ;;
+    esac
 
-# Disable ccache on darwin since the brew install step fails.
-#install:
-#  - if [ "$TRAVIS_OS_NAME" = "osx" ]; then brew install ccache; fi
-#  - if [ "$TRAVIS_OS_NAME" = "osx" ]; then export PATH="/usr/local/opt/ccache/libexec:$PATH"; fi
+before_cache:
+- |-
+    case $TRAVIS_OS_NAME in
+      windows)
+        # https://unix.stackexchange.com/a/137322/107554
+        $msys2 pacman --sync --clean --noconfirm
+        ;;
+    esac
+
+cache:
+    directories:
+    - $HOME/AppData/Local/Temp/chocolatey
+    - /C/tools/msys64
 
 matrix:
   include:
   - os: linux
     name: "MPI-Linux SMP"
     dist: xenial
-    env: CACHE_NAME=mpi-linux-smp
     script:
-    - sudo apt-get install -y mpich libmpich-dev
+    - sudo apt-get update && sudo apt-get install -y mpich libmpich-dev
     - ./build LIBS mpi-linux-x86_64 smp -j3 --with-production --enable-error-checking --enable-lbuserdata -g -Werror=vla || exit 1
     - make -C mpi-linux-x86_64-smp/tmp -j3 all-test-tests all-test-examples OPTS="-g"
     - make -C mpi-linux-x86_64-smp/tmp test-tests test-examples TESTOPTS="+setcpuaffinity"
+
   - os: osx
     name: "NetLRTS-Darwin"
-    env: CACHE_NAME=netlrts-darwin
     script:
     - ./build LIBS netlrts-darwin-x86_64 -j3 --with-production --enable-error-checking --enable-lbuserdata -g -Werror=vla || exit 1
     - make -C netlrts-darwin-x86_64/tmp -j3 all-test-tests all-test-examples OPTS="-g"
     - make -C netlrts-darwin-x86_64/tmp test-tests test-examples TESTOPTS="++local"
- # Windows build does not work yet due to missing autoconf/automake
- # - os: windows
- #   name: "NetLRTS-Win"
- #   script:
- #   - choco install -y msys2
- #   - pacman -S autoconf automake
- #   - ./build LIBS netlrts-win-x86_64 -j2 -k --without-romio --with-production --enable-error-checking --enable-lbuserdata -g || exit 1
- #   - make -C netlrts-win-x86_64/tmp -k -j2 all-test-tests all-test-examples OPTS="-g"
- #   - make -C netlrts-win-x86_64/tmp -k test-tests test-examples TESTOPTS="++local"
+
+  - os: windows
+    name: "NetLRTS-Win"
```

Comment:
Would recommend naming this something like "NetLRTS-Win GCC" to distinguish it from the default MSVC builds

Reply:
fixed.